app/src/main/java/com/github/se/cyrcle/ui/theme/molecules/TopAppBar.kt

```diff
@@ -49,10 +61,57 @@ fun TopAppBar(navigationActions: NavigationActions, title: String, testTag: Stri
               scrolledContainerColor = MaterialTheme.colorScheme.primaryContainer),
       modifier = Modifier.testTag(testTag),
       navigationIcon = {
-        IconButton(
-            onClick = { navigationActions.goBack() },
-            modifier = Modifier.testTag("${testTag}GoBackButton")) {
-              Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Go Back")
-            }
+        Box(modifier = Modifier.padding(8.dp)) {
+          IconButton(
+              onClick = { navigationActions.goBack() },
+              modifier =
+                  Modifier.testTag("${testTag}GoBackButton")
+                      .background(Color.White.copy(alpha = 0.1f), shape = RoundedCornerShape(12.dp))
+                      .size(40.dp)) {
+                Icon(
+                    Icons.AutoMirrored.Filled.ArrowBack,
+                    contentDescription = "Go Back",
+                    modifier = Modifier.size(24.dp))
+              }
+        }
+      })
+}
+
+/**
+ * A composable function that manages the resizing of the text to fit the available space. The text
+ * will be resized until it fits the available space, or until the font size is 6sp. Both the font
+ * size and the line height will be reduced by 5% at each iteration.
+ *
+ * @param text The text to display
+ * @param initialFontSize The initial font size of the text
+ * @param modifier The modifier of the text
+ * @param maxLines The maximum number of lines the text can take
+ */
+@Composable
+fun AutoResizingText(
```

Comment:
Modularization makes the code a lot clearer well done !

Reply:
Thank you!